src/app/clusters/power-source-server/power-source-server.h

```diff
@@ -0,0 +1,91 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/util/af-types.h>
+#include <app/util/basic-types.h>
+#include <lib/support/Span.h>
+#include <platform/CHIPDeviceConfig.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+struct PowerSourceClusterInfo
+{
+    PowerSourceClusterInfo() : mClusterEndpoint(kInvalidEndpointId) {}
+    explicit PowerSourceClusterInfo(EndpointId powerClusterEndpointId) : mClusterEndpoint(powerClusterEndpointId) {}
+    CHIP_ERROR SetEndpointList(Span<EndpointId> endpointList)
+    {
+        if (endpointList.size() == 0)
+        {
+            mBuf          = nullptr;
+            mEndpointList = Span<EndpointId>();
+            return CHIP_NO_ERROR;
+        }
+
+        mBuf = std::make_unique<EndpointId[]>(endpointList.size());
+        if (mBuf == nullptr)
```

Comment:
I think on oom make_unique called like this is going to crash (or at least try to throw an exception, then crash because they are disabled), not return null.  We probably need to use ScopedMemoryBuffer here, if we go with heap allocation.

Reply:
+1 to ScopedMemoryBuffer

Reply:
Changed.